src/Compilation/Config.zig

```diff
@@ -302,15 +304,15 @@ pub fn resolve(options: Options) ResolveError!Config {
             .ilp32,
             .muslabin32,
             .muslx32,
-            => break :b false,
+            => break :b .none,
             else => {},
         }
 
         break :b switch (options.output_mode) {
-            .Lib, .Obj => false,
+            .Lib, .Obj => .none,
             .Exe => switch (root_optimize_mode) {
-                .Debug => false,
-                .ReleaseSafe, .ReleaseFast, .ReleaseSmall => true,
+                .Debug => .none,
+                .ReleaseSafe, .ReleaseFast, .ReleaseSmall => options.lto,
```

Comment:
In this path, we're supposed to resolve a default value if the user has not provided one. We already checked `options.lto != .none` above, so this can only possibly resolve to `.none`.

This should probably be:

```suggestion
                .ReleaseSafe, .ReleaseFast, .ReleaseSmall => .full,
```

We can consider later if the default should be switched to `.thin`. I don't know if there are any major disadvantages to it that would make `.full` a preferable default.

Reply:
@TCROC ping; I think this hunk is the only blocker for merging this.